bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/item/ItemResource.java

```diff
@@ -641,12 +647,17 @@ public Response createOrUpdateItems(
         Map<String, Collection<String>> tagMap = new HashMap<>();
 
         for (GroupItemDTO item : items) {
-            Item newItem = ItemDTOMapper.map(item, itemBuilderFactory);
-            if (newItem == null) {
-                wrongTypes.add(item);
-                tagMap.put(item.name, item.tags);
-            } else {
-                activeItems.add(newItem);
+            try {
+                Item newItem = ItemDTOMapper.map(item, itemBuilderFactory);
+                if (newItem == null) {
+                    wrongTypes.add(item);
+                    tagMap.put(item.name, item.tags);
+                } else {
+                    activeItems.add(newItem);
+                }
+            } catch (IllegalArgumentException e) {
+                logger.warn("Received HTTP PUT request with an invalid item name '{}'.", item.name);
+                return Response.status(Status.BAD_REQUEST).build();
```

Comment:
The OpenAPI doc should be improved for this response since this response is currently unexpected for non-null item lists:

```java
@ApiResponse(responseCode = "400", description = "Item list is null.")
```